### PR TITLE
Fix Cashflow by service report and search modal for caution in payment registry

### PR DESCRIPTION
**Pull request by mbayopanda**
This PR fix the issue #3819 

And enhance the search modal in payment registry for caution by allowing to filter payments which are caution or payments which are not caution.

### CODE REVIEW

File: server/controllers/finance/reports/cashflow/index.js [hunk 1]
```diff
@@ -117,7 +117,7 @@ function reportByService(req, res, next) {
   db.one(cashboxDetailsSql, cashboxId)
```
Comment (jniles) [inline server/controllers/finance/reports/cashflow/index.js]: !!!  Can we rename this variable from `cashboxId` to `cashboxAccountId`?  The origin of this bug seems to be from a bad name.

**Review by jniles — APPROVED** [review]
Looks good to me.  Great catch!

bors r+

**Comment by jniles** [comment]
Welp.

bors r+

File: client/src/modules/templates/bhYesNoRadios.tmpl.html [hunk 2]
```diff
@@ -12,7 +14,7 @@
         ng-value="1"
         ng-model="$ctrl.value"
         ng-change="$ctrl.onChange($ctrl.value)"
-        required>
+        ng-required="$ctrl.required">
```
Comment (jniles) [inline client/src/modules/templates/bhYesNoRadios.tmpl.html]: Why do we need to toggle the required if we have a `bhClear`?  It seems like both of them are addressing the same solution.

Comment (mbayopanda) [reply client/src/modules/templates/bhYesNoRadios.tmpl.html]: You're right @jniles

**Review by jniles — APPROVED** [review]
Looks good to me!

**Comment by mbayopanda** [comment]
@jniles could review this PR please

**Comment by jniles** [comment]
I did!  And I approved it.  You can merge anytime.

**Comment by mbayopanda** [comment]
Thanks @jniles

bors r+

**Pull request merged**